controllers/artists.js

```diff
@@ -27,37 +26,318 @@ const handleErrorAsync = (fn) => async (req, res, next) => {
 const get_artist_by_url = handleErrorAsync(async (req, res, next) => {
   const request = {
     artistUrl: req.params.artistUrl,
-    // limit: req.query.limit ? req.query.limit : 10,
-    // sortBy: req.body.sortBy
   };
 
-  if (request.artistUrl) {
-    db.knex("artist")
-      .select(
-        "artist.id as id",
-        "artist.name as name",
-        "artist.bio as bio",
-        "artist.twitter as twitter",
-        "artist.npub as nostr",
-        "artist.instagram as instagram",
-        "artist.youtube as youtube",
-        "artist.website as website",
-        "artist.artwork_url as avatarUrl",
-        "artist.verified as verified",
-        "artist.updated_at as updateDate"
+  const artist = await prisma.artist.findFirstOrThrow({
+    where: { artist_url: request.artistUrl },
+  });
+
+  res.json(artist);
+});
+
+const get_artist_by_id = handleErrorAsync(async (req, res, next) => {
+  const request = {
+    artistId: req.params.artistId,
+  };
+
+  const artist = await prisma.artist.findFirstOrThrow({
+    where: { id: request.artistId },
+  });
+
+  res.json(artist);
+});
+
+const create_artist = handleErrorAsync(async (req, res, next) => {
+  const newArtistId = randomUUID();
+
+  // TODO: Add other attributes like twitter handle, etc.
+  const request = {
+    artwork: req.file,
+    userId: req.uid, //required, should come in with auth
+    name: req.body.name, // required
+    bio: req.body.bio,
+  };
+
+  let uploadPath;
+  let isKeeper = false;
+  if (!request.artwork) {
+    uploadPath = "./graphics/wavlake-icon-750.png";
+    isKeeper = true;
+  } else {
+    uploadPath = request.artwork.path;
+  }
+
+  // console.log(request.image)
+  // console.log(uploadPath)
+
+  const convertPath = `${localConvertPath}/${newArtistId}.jpg`;
+  const s3Key = `${imagePrefix}/${newArtistId}.jpg`;
+
+  Jimp.read(uploadPath)
+    .then((img) => {
+      return img.resize(1875, Jimp.AUTO).quality(70).writeAsync(convertPath); // save
+    })
+    // Upload to S3
+    .then((img) => {
+      s3Client
```

Comment:
i think this should be returned
```suggestion
      return s3Client
```